CorsixTH/Lua/entities/humanoid.lua

```diff
@@ -637,7 +637,7 @@ function Humanoid:_handleEmptyActionQueue()
       else
         -- Set these variables to increase the likelihood of the humanoid managing to get out of the hospital.
         self.going_home = false
-        self.hospital = self.world:getLocalPlayerHospital()
+        self.hospital = self.world:getHospital(self.tile_x, self.tile_y)
```

Comment:
Bit nervous about this now because we're already in a dicey place with an empty action queue. If the humanoid isn't on a hospital tile this would just crash.
![image](https://github.com/CorsixTH/CorsixTH/assets/20030128/0df9ec44-9cc1-4684-b44a-ddd3f5bac4e4)